src/components/UserProfile/ContactReadOnly.tsx

```diff
@@ -0,0 +1,158 @@
+import React from 'react';
+import { Card } from 'antd';
+
+import { LoggedInUser } from 'store/userTypes';
+
+import CardNoDataReadOnly from './CardNoDataReadOnly';
+import ContactReadInformation from './ContactReadInformation';
+import FindMeReadOnly from './FindMeReadOnly';
+import {
+  extractFindMeFromProfile,
+  extractRoleFromProfile,
+  isResearcher,
+  makeCommonCardPropsReadOnly,
+  showInstitution,
+  showWhenHasDataOrCanEdit,
+} from './utils';
+
+import './style.css';
+
+type Props = {
+  data: LoggedInUser;
+  canEdit: boolean;
+  isProfileUpdating: boolean;
+  onClickEditCb: () => void;
+};
+
+const getInstitutionLabelGivenRole = (data: LoggedInUser) => {
+  const role = extractRoleFromProfile(data);
+  if (role === 'research') {
+    return 'Institution';
+  } else if (role === 'community') {
+    return 'Institution/Organization';
+  }
+  return '';
+};
+
+const COMMUNITY_SPECIFIC_PROPERTIES = ['institution', 'department', 'institutionalEmail'];
+
+const RESEARCHER_SPECIFIC_PROPERTIES = [
+  'jobTitle',
+  'institution',
+  'department',
+  'institutionalEmail',
+];
+const COMMON_PROPERTIES = [
+  'email',
+  'zip',
+  'phone',
+  'addressLine1',
+  'addressLine2',
+  'city',
+  'state',
+  'country',
+];
+
+const hasData = (data: LoggedInUser) => {
+  if (!data) {
+    return false;
+  }
+  const role = extractRoleFromProfile(data);
+  let fieldsToCheckFor = COMMON_PROPERTIES;
+  if (role === 'research') {
+    fieldsToCheckFor = fieldsToCheckFor.concat(RESEARCHER_SPECIFIC_PROPERTIES);
+  } else if (role === 'community') {
+    fieldsToCheckFor = fieldsToCheckFor.concat(COMMUNITY_SPECIFIC_PROPERTIES);
+  }
+  return fieldsToCheckFor.some((f) => data[f as keyof LoggedInUser]);
```

Comment:
In this case, the goal is to verify if the data received are empty. Therefore, I'm not sure about the benefits and will not modify the code. However, you have a good point and I think it would be nice to construct a User object with utility functions (some that would be related to the `role`) that would simply some logic used a crossed the portal.